src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -762,7 +912,122 @@ inline ${optionalType}<T> get_optional(const json &j, const char *property) {
             });
         }
 
-        this.ensureBlankLine();
-        this.emitLine("#endif");
+        this.finishFile();
+    }
+
+    protected emitClassDefinition(c: ClassType, className: Name): void {
+        this.startFile(this.sourcelikeToString(className)+".hpp");
+        /** 
+         * Need to generate "imports", in terms 'c' has members, which
+         * are defined by others
+         */
+        let included: boolean = false;
+        this.forEachClassProperty(c, "none", (_name, _jsonName, property) => {
+
+            let p = property.type;
+
+            if (p instanceof ArrayType) {
+                p = p.items;
+            }
+
+            const propertyType = this.sourcelikeToString(this.cppType(p, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true));
+            if (p instanceof ClassType ||
+                p instanceof EnumType ||
+                p instanceof UnionType) {
+
+                this.forEachDeclaration("none", decl => {
```

Comment:
This whole way of doing this seems very roundabout and fickle.  I don't have a suggestion off the top of my head how to fix it, and I'm happy to merge it as it is.  Just telling myself to be prepared for this to break, really.